src/apps/vpn/platforms/ios/iostunnel.swift

```diff
@@ -73,6 +72,9 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
                 self.logger.info(message: "Tunnel interface is \(interfaceName)")
 
                 if self.isSuperDooperFeatureActive {
+                    GleanMetrics.Session.daemonSessionSource.set(isSourceApp ? "app" : "system settings")
+                    GleanMetrics.Session.daemonSessionId.generateAndSet()
+                    GleanMetrics.Session.daemonSessionStart.set()
```

Comment:
This implementation will only record successful sessions, is that alright? Not sure we discussed this in the proposal i.e. if a session fails to start we won't see it in the data. This is not necessarily a problem, just a question.

Reply:
The intention is to record successful sessions, and I believe this implementation does that.